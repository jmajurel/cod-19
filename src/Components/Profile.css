.formGroup {
  margin: 2rem 0;
}
.profile label {
  display: block;
}
