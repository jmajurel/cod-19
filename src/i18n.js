import i18n from "i18next";
import { initReactI18next } from "react-i18next";

import LanguageDetector from "i18next-browser-languagedetector";
import frenchTranslation from "./translations/french.json";
import englishTranslation from "./translations/english.json";
// the translations
// (tip move them in a JSON file and import them)
const resources = {
  en: englishTranslation,
  fr: frenchTranslation
};

i18n
  // detect user language
  .use(LanguageDetector)
  // pass the i18n instance to react-i18next.
  .use(initReactI18next)
  // init i18next
  .init({
    lng: "fr",
    fallbackLng: "en",
    resources,
    ns: ["moduleA", "moduleB"],
    interpolation: {
      escapeValue: false // not needed for react as it escapes by default
    }
  });
export default i18n;
