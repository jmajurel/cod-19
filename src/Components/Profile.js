import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import Address from "./Address";

import "./Profile.css";

const Profile = ({ existingProfile, specialities, onSubmit }) => {
  const [firstName, setFirstName] = useState(existingProfile.firstName);
  const [lastName, setLastName] = useState(existingProfile.lastName);
  const [speciality, setSpeciality] = useState({
    id: existingProfile.speciality,
    data: specialities.find((x) => x._id === existingProfile.speciality),
  });
  const [address, setAddress] = useState(existingProfile.address);
  let history = useHistory();

  function handleSubmit(event) {
    event.preventDefault();
    onSubmit({
      email: existingProfile.email,
      firstName,
      lastName,
      address,
      speciality: speciality.id,
    });
    history.push("/home");
  }

  function handleAddressChange(newAddress) {
    setAddress(newAddress);
  }

  function handleSpecialityChange(event) {
    const selectedSpeciality = event.target.value;
    setSpeciality({
      id: selectedSpeciality,
      data: specialities.find((x) => x._id === selectedSpeciality),
    });
  }

  return (
    <div className="profile">
      <h2>My Profile</h2>
      <form onSubmit={handleSubmit}>
        <div className="formGroup profileImg"></div>
        <div className="formGroup">
          <label>Email: </label>
          <input
            disabled
            name="email"
            type="text"
            value={existingProfile.email}
          />
        </div>
        <div className="formGroup">
          <label>First name: </label>
          <input
            name="firstName"
            type="text"
            value={firstName}
            onChange={(e) => setFirstName(e.target.value)}
          />
        </div>

        <div className="formGroup">
          <label>Last name: </label>
          <input
            name="lastName"
            type="text"
            value={lastName}
            onChange={(e) => setLastName(e.target.value)}
          />
        </div>

        <div className="formGroup">
          <label>Speciality: </label>
          <select value={speciality.id} onChange={handleSpecialityChange}>
            {specialities &&
              specialities.map((speciality) => (
                <option key={speciality._id} value={speciality._id}>
                  {speciality.name}
                </option>
              ))}
          </select>
        </div>
        <Address address={address} onChange={handleAddressChange} />
        <button type="submit">Save</button>
      </form>
    </div>
  );
};

export default Profile;
