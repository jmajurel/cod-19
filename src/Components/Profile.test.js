import React from "react";
import Profile from "./Profile";
import { render, unmountComponentAtNode } from "react-dom";
import { MemoryRouter } from "react-router-dom";
import { I18nextProvider } from "react-i18next";
import renderer from "react-test-renderer";
import i18n from "../i18n";
import { act } from "react-dom/test-utils";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("renders correctly", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <I18nextProvider i18n={i18n}>
          <Profile />
        </I18nextProvider>
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders with a profile", () => {
  const mockProfile = { firstName: "John" };
  act(() => {
    render(
      <MemoryRouter>
        <I18nextProvider i18n={i18n}>
          <Profile existingProfile={mockProfile} />
        </I18nextProvider>
      </MemoryRouter>,
      container
    );
  });
  const input = container.querySelector("input[name=firstName]");
  expect(input.value).toBe(mockProfile.firstName);
});

it("reacts on submit", () => {
  const onSubmit = jest.fn();
  const mockProfile = { email: "john.doe@gmail.com", firstName: "John" };
  act(() => {
    render(
      <MemoryRouter>
        <I18nextProvider i18n={i18n}>
          <Profile existingProfile={mockProfile} onSubmit={onSubmit} />
        </I18nextProvider>
      </MemoryRouter>,
      container
    );
  });

  const submitBtn = document.querySelector("button[type=submit]");

  //simulate user click on save btn
  act(() => {
    submitBtn.dispatchEvent(new MouseEvent("click", { bubbles: true }));
  });
  expect(onSubmit).toHaveBeenCalledTimes(1);
});
