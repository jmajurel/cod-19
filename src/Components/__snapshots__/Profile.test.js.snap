// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="profile"
>
  <h2>
    Mon Profil
  </h2>
  <form
    onSubmit={[Function]}
  >
    <div
      className="formGroup profileImg"
    />
    <div
      className="formGroup"
    >
      <label>
        Email
        : 
      </label>
      <input
        disabled={true}
        name="email"
        type="text"
      />
    </div>
    <div
      className="formGroup"
    >
      <label>
        Prénom
        : 
      </label>
      <input
        name="firstName"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="formGroup"
    >
      <label>
        Nom
        : 
      </label>
      <input
        name="lastName"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="formGroup"
    >
      <label>
        Spécialité médicale
        : 
      </label>
      <select
        onChange={[Function]}
      />
    </div>
    <div
      className="address"
    >
      <div
        className="formGroup"
      >
        <label>
          Adresse
          : 
        </label>
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="formGroup"
      >
        <label>
          Complément d'adresse
          : 
        </label>
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="formGroup"
      >
        <label>
          Ville
          : 
        </label>
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="formGroup"
      >
        <label>
          Code postal
          : 
        </label>
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="formGroup"
      >
        <label>
          Région
          : 
        </label>
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="formGroup"
      >
        <label>
          Pays
          : 
        </label>
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <button
      type="submit"
    >
      Enregistrer
    </button>
  </form>
</div>
`;
